zitadel/v2/domain/resource.go

```diff
@@ -27,7 +27,7 @@ func GetResource() *schema.Resource {
 			},
 			isPrimaryVar: {
 				Type:        schema.TypeBool,
-				Computed:    true,
+				Optional:    true,
```

Comment:
So just for my understanding:
Terraform replaces the resource automatically, aka deletes and creates it, if we don't specify an update callback? Right?

Reply:
you're right, added an update function which only triggers when isprimary is changed and sets primary if true

Reply:
But that disables changing a domain name, right? Replacing is probably better?

Reply:
there is no changing a domain name as the domain name is the primary key, and I created the update function as the primary attribute is separate and if you change the boolean you have to call ane endpoint

Reply:
in a terraform resources perspective, we could replace a domain you implemented initially, effectively enabling changing a domain name of a specific terraform resource.
But fine by me if we decide against it.